### ISSUE TITLE
Progress indicator

### ISSUE THREAD

**Issue by Dynom** [phpmd/phpmd#94]
The closest I came was issue https://github.com/phpmd/phpmd/issues/37 but I'm unsure if that request is identical to this one.

We run PHPMD frequently (it's part of a pre-push hook) but it's very inconvenient that no progress, at all, is shown by PHPMD. We'll settle for anything as trivial as e.g. an output like PHPUnit (just a dot per test) or fancy, a 0-100% indicator but either way. Some form of progress-feedback is highly appreciated.

**Comment by JeroenDeDauw** [comment]
+1 to progress reporting. Has any work on this been done? Looks like b94cb21 incorrectly referenced this issue

**Comment by atulatri** [comment]
+1

**Comment by fezfez** [comment]
+1

**Comment by schmengler** [comment]
+1

**Comment by quentincurtet** [comment]
+1

**Comment by keradus** [comment]
:+1:

**Comment by tabishjaved03** [comment]
+1

**Comment by josephdpurcell** [comment]
+1

**Comment by alfredbez** [comment]
Please use [GitHub Reactions](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments)

**Comment by josephdpurcell** [comment]
I think #37 would be an acceptable way to show progress.

To me, a lack of logging/debug reporting is itself a bug. PHPMD is an important part of the PHP software community, and if something goes wrong and there's no error/indication what is going on, you cannot use PHPMD. I just ran PHPMD and after 104 mins I still have no sign of life from PHPMD.

I am working on debugging PHPMD timeouts on Code Climate, see https://github.com/codeclimate/codeclimate-phpmd/issues/30--this feature would be very helpful!

**Comment by PierreRambaud** [comment]
+1

**Comment by mikepmtl** [comment]
Part of our Jenkins build process.  No idea if it is running or not.  Even just a "." would be nice.
Thank you.

**Comment by orlangur** [comment]
> I just ran PHPMD and after 104 mins I still have no sign of life from PHPMD.

I don't think this is a valid use case :) The problem here is that some project-wide inspections are waaay too slow. Maybe 10 minutes is too strict for build to complete but it should not take hours.

Is PHPMD currently optimal in terms of computational complexity? Some `O(N*N)` where N is lines of code in project?`

**Comment by akielbasa** [comment]
i found an issue with user rights on windows domains, this might lead to an issue with phpmd beeing super slow

correcting the speed on my mashine from 20minute precommit
to 20 second precommit

the case is:

while using phpstorm (but for console its the same)
having the project outside your user directory

starting phpstorm / cmd as admin (yes you NEED to be lokal admin for this)
fixing the issue 

i was able to see this issue on a lot of pc's in our company

i know this has nothing to do with the progressbar which would be a blast,
but most people complain that php md is actually too slow raising up to 104 minutes of precommits,

if this is your case, you might not have this issue in the future anymore


tl;dr

if you have this scenario:
your project ist not in your userfolder
try running console / phpstorm as admin (local admin rights needed when in domain)
see if the time is better

on our case
20 minutes => 12~ seconds

**Comment by InvisibleSmiley** [comment]
> i found an issue with user rights on windows domains, this might lead to an issue with phpmd beeing super slow

Thanks for the pointer, I had the same problem! Further investigation showed that this is actually a PHP Depend issue. Filed https://github.com/pdepend/pdepend/issues/447

Workaround: Copy https://github.com/pdepend/pdepend/blob/master/pdepend.xml.dist to the root of your project, optionally naming it either pdepend.xml

**Comment by tebeso** [comment]
i can confirm that running phpmd on windows is super slow (15 seconds for one file) and when i run as admin, its 44ms. Please fix this.

**Comment by kylekatarnls** [comment]
This is not happening on every Windows machine. It's happening on particular settings having `HOME` set on a slow file system/slow network. It's not actually PHPMD / PDepend being slow, but `HOME` read/write, you can solve it by setting a custom `HOME` (to some directory with a faster access) when running the command.

**Comment by tebeso** [comment]
> 
> 
> This is not happening on every Windows machine. It's happening on particular settings having `HOME` set on a slow file system/slow network. It's not actually PHPMD / PDepend being slow, but `HOME` read/write, you can solve it by setting a custom `HOME` (to some directory with a faster access) when running the command.

and how do i do that?

**Comment by kylekatarnls** [comment]
```
set HOME=C:\Path\To\A\Fast\Directory
phpmd src text phpmd.xml
```

**Comment by prestamod** [comment]
> set HOME=C:\Path\To\A\Fast\Directory
I love you

**Comment by AJenbo** [comment]
This is part of https://github.com/phpmd/phpmd/pull/1256

**Comment by keradus** [comment]
awesome news!
